server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java

```diff
@@ -67,84 +66,79 @@ public void testExtensionsDiscovery() throws Exception {
             "     hostName: 'myIndependentPluginHost1'",
             "     hostAddress: '127.0.0.0'",
             "     port: '9300'",
-            "     version: '3.0.0'",
+            "     version: '1.0'",
+            "     description: Fake description 1",
+            "     opensearchVersion: '3.0.0'",
+            "     javaVersion: '14'",
+            "     className: fakeClass1",
+            "     customFolderName: fakeFolder1",
+            "     hasNativeController: false",
             "   - name: secondExtension",
             "     uniqueId: 'uniqueid2'",
             "     hostName: 'myIndependentPluginHost2'",
             "     hostAddress: '127.0.0.1'",
             "     port: '9301'",
-            "     version: '2.0.0'"
+            "     version: '1.1'",
+            "     description: Fake description 2",
+            "     opensearchVersion: '2.0.0'",
+            "     javaVersion: '17'",
+            "     className: fakeClass2",
+            "     customFolderName: fakeFolder2",
+            "     hasNativeController: true"
         );
         Files.write(extensionDir.resolve("extensions.yml"), extensionsYmlLines, StandardCharsets.UTF_8);
 
-        Path pluginDir1 = extensionDir.resolve("firstExtension");
-        Path pluginDir2 = extensionDir.resolve("secondExtension");
-
-        PluginTestUtil.writePluginProperties(
-            pluginDir1,
-            "description",
-            "fake desc",
-            "name",
-            "firstExtension",
-            "version",
-            "1.0",
-            "opensearch.version",
-            Version.CURRENT.toString(),
-            "java.version",
-            System.getProperty("java.specification.version"),
-            "classname",
-            "FakeExtensions"
-        );
-
-        PluginTestUtil.writePluginProperties(
-            pluginDir2,
-            "description",
-            "fake desc",
-            "name",
-            "secondExtension",
-            "version",
-            "1.0",
-            "opensearch.version",
-            Version.CURRENT.toString(),
-            "java.version",
-            System.getProperty("java.specification.version"),
-            "classname",
-            "FakeExtensions"
-        );
-
         Settings settings = Settings.builder().build();
         ExtensionsOrchestrator extensionsOrchestrator = new ExtensionsOrchestrator(settings, extensionDir);
 
-        Set<DiscoveryExtension> expectedExtensionsSet = new HashSet<DiscoveryExtension>();
+        List<DiscoveryExtension> expectedExtensionsList = new ArrayList<DiscoveryExtension>();
 
-        expectedExtensionsSet.add(
+        expectedExtensionsList.add(
             new DiscoveryExtension(
                 "firstExtension",
-                "id",
+                "uniqueid1",
```

Comment:
Actually a serious comment!  Are ID's envisioned to be text-based like this, or in UUID format, or what?  The example here should probably mimic our vision.